primitives/src/rule.rs

```diff
@@ -0,0 +1,54 @@
+use crate::{Claim, ClaimType, IdentityId};
+use codec::{Decode, Encode};
+use sp_std::prelude::*;
+
+#[derive(Encode, Decode, Clone, PartialEq, Eq, Debug)]
+/// It defines the type of rule supported, and the filter information we will use to evaluate as a
+/// predicate.
+pub enum RuleType {
+    /// Rule to ensure that claim filter produces one claim.
+    IsPresent(ClaimType),
+    /// Rule to ensure that claim filter produces an empty list.
+    IsAbsent(ClaimType),
+    /// Rule to ensure that at least one claim is fetched when filter is applied.
+    IsAnyOf(Vec<Claim>),
+}
```

Comment:
I think we also need `IsNoneOf(Vec<Claim>)`.

Reply:
`IsNoneOf` has been added.
The predicate implementation is the following: 
```
RuleType::IsNoneOf(claims) => not(any(claims)).evaluate(context),
```